src/NHibernate/Cache/ReadOnlyCache.cs

```diff
@@ -87,7 +87,7 @@ public ISoftLock Lock(CacheKey key, object version)
 				return result;
 			}
 
-			var checkKeys = new List<CacheKey>();
+			var checkKeys = new List<object>();
```

Comment:
This change seems unnecessary (same for async version)

Reply:
This change results from the merge, not from a manual change during conflict resolution.
Reverting it would mean reverting a part of a change done in 5.2.6.
This change comes from [here](https://github.com/nhibernate/nhibernate-core/commit/c9ab94d315b56dc913745592ff052481f034f2a8#diff-cc4750cddf1d3379d1c9ea40b4256323R94), it has been done to avoid a cast or an array type coercion. It is not unnecessary, and already validated by #2128 review.

Reply:
As you can see there is no `Select(o => (object) o)` removal on master because it's already removed [here](https://github.com/nhibernate/nhibernate-core/commit/5ee4401702a4f8eeddc05c7f09646bf14ea3ecdd#diff-cc4750cddf1d3379d1c9ea40b4256323R58) in #2161 without this change. So this change is not necessary on master and could be removed by squashing it in merge commit.

Reply:
Ok, done.